nixos/modules/services/web-servers/nginx/cgit.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nginx.cgit;
+  configFile = pkgs.writeText "cgitrc"
+    ''
+      css=/cgit.css
+      logo=/cgit.png
+      favicon=/favicon.ico
+
+      # user configuration
+      ${cfg.configText}
+    '';
+in
+{
+
+  options.services.nginx.cgit = {
+
+    enable = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        If true, enable cgit (fast web interface for git repositories) as a
+        sub-service in nginx.
+      '';
+    };
+
+    location = mkOption {
+      default = "/";
+      type = types.str;
+      description = ''
+        Location to serve cgit on.
+      '';
+    };
+
+    configText = mkOption {
+      default = "";
+      example = ''
+        virtual-root=/
+        source-filter=''${pkgs.cgit}/lib/cgit/filters/syntax-highlighting.py
+        about-filter=''${pkgs.cgit}/lib/cgit/filters/about-formatting.sh
+        cache-size=1000
+        scan-path=/srv/git
+      '';
+      type = types.lines;
+      description = ''
+        Verbatim contents of the cgit runtime configuration file. Documentation
+        (with cgitrc example file) is available in "man cgitrc". Or online:
+        http://git.zx2c4.com/cgit/tree/cgitrc.5.txt
+      '';
+    };
```

Comment:
This way of configuring services is discouraged (see [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md)).  Since the config file has an INI-like format, it should be easy to generate it from Nix expressions.  See also https://nixos.org/manual/nixos/stable/index.html#sec-freeform-modules